driver-core/src/main/com/mongodb/connection/TransportSettings.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.connection;
+
+import com.mongodb.connection.netty.NettySettings;
+
+/**
+ * Transport settings for the driver.
+ *
+ * @since 4.11
+ */
+public abstract class TransportSettings {
```

Comment:
I envision `GrpcSettings extends TransportSettings` in the future. Even if `GrpcSettings` does not bear any settings in the beginning or ever, specifying `GrpcSettings` will be the way to enable gRPC via `MongoClientSettings`. How does this seem to you?

Reply:
I anticipated the same.  The only weird part perhaps is that, since the Java gRPC client uses Netty as its I/O substrate, we may want to embed NettySettings in GrpcSettings.

Reply:
If we can support them all (I don't remember), that may be helpful. We'll see.